src/test/scala/chiselTests/RecordSpec.scala

```diff
@@ -132,6 +132,39 @@ trait RecordSpecUtils {
     out := in1 + in2
   }
 
+  class InnerRecord extends Record {
+    val k = new InnerInnerRecord
+    val elements = SeqMap("" -> k)
+    override def opaqueType = elements.size == 1
+    override def cloneType: this.type = (new SingleElementRecord).asInstanceOf[this.type]
+  }
+
+  class InnerInnerRecord extends Record {
+    val k = new SingleElementRecord
+    val elements = SeqMap("" -> k)
+    override def opaqueType = elements.size == 1
+    override def cloneType: this.type = (new SingleElementRecord).asInstanceOf[this.type]
+  }
+
+  class NestedRecordModule extends Module {
+    val in1 = IO(Input(new InnerRecord))
+    val out = IO(Output(new InnerRecord))
+
+    out := in1
+
+    // DO NOT do this, just for testing element connection
+    out.elements.head._2 := in1.elements.head._2
+
+    val k = Module(new InnerModule)
+  }
+
+  class InnerModule extends Module {
+    val in1 = IO(Input(new InnerRecord))
+    val out = IO(Output(new InnerRecord))
+
+    out := in1
+  }
```

Comment:
```suggestion
  class NestedRecordModule extends Module {
    val in = IO(Input(new InnerRecord))
    val out = IO(Output(new InnerRecord))

    val inst = Module(new InnerModule)
    inst.foo := in
    out := inst.bar
  }

  class InnerModule extends Module {
    val foo = IO(Input(new InnerRecord))
    val bar = IO(Output(new InnerRecord))
    
    // DO NOT do this, just for testing element connection
    bar.elements.head._2 := foo.elements.head._2

  }
```
You have the connections, but aren't actually checking for them down below. With these suggestions, I would add a check in the chirrtl for the following lines:
```
input in : UInt<8>
output out : UInt<8>
inst.foo <= in
out <= inst.bar
input foo : UInt<8>
output bar : UInt<8>
bar <= foo
```